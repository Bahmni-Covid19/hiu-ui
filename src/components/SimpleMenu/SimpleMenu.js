import React, { useEffect } from "react";
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";

const SimpleMenu = ({ menuItems, selectedValue, handleChange }) => {
  return (
    <Select id="select-menu" value={selectedValue} onChange={handleChange}>
      {menuItems.map(type => (
        <MenuItem
          id="menu-item"
          key={type.value}
          value={type.value}
          onClick={handleChange}
        >
          {type.label}
        </MenuItem>
      ))}
    </Select>
  );
};

export default SimpleMenu;
