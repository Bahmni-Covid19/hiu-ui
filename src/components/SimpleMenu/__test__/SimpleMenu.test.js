import { shallow } from 'enzyme';
import React from 'react';
import SimpleMenu from '../SimpleMenu';

describe('Simple Menu', () => {
  const MenuItems = [
    {
      label: 'General Consulting',
      value: 'GeneralConsulting',
    },
    {
      label: 'Referral services',
      value: 'ReferralServices',
    },
    {
      label: 'Episode of Care',
      value: 'EpisodeOfCare',
    },
    {
      label: 'Encounter',
      value: 'Encounter',
    },
  ];
  it('should render properly ', () => {
    const wrapper = shallow(
      <SimpleMenu menuItems={MenuItems} selectedValue="GeneralConsulting" />,
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render properly', () => {
    const noop = jest.fn();
    const wrapper = shallow(
      <SimpleMenu
        menuItems={MenuItems}
        selectedValue="GeneralConsulting"
        handleChange={noop}
      />,
    );

    wrapper.find('#select-menu').simulate('click');

    expect(noop).toHaveBeenCalled;
  });
});
