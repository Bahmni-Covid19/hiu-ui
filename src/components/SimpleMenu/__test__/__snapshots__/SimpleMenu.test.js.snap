// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Menu should render properly  1`] = `
"<WithStyles(ForwardRef(Select)) id=\\"select-menu\\" value=\\"GeneralConsulting\\" onChange={[undefined]}>
  <WithStyles(ForwardRef(MenuItem)) id=\\"menu-item\\" value=\\"GeneralConsulting\\" onClick={[undefined]}>
    General Consulting
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem)) id=\\"menu-item\\" value=\\"ReferralServices\\" onClick={[undefined]}>
    Referral services
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem)) id=\\"menu-item\\" value=\\"EpisodeOfCare\\" onClick={[undefined]}>
    Episode of Care
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem)) id=\\"menu-item\\" value=\\"Encounter\\" onClick={[undefined]}>
    Encounter
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Select))>"
`;
